modyn/selector/internal/selector_strategies/abstract_downsample_strategy.py

```diff
@@ -23,11 +24,44 @@ class AbstractDownsampleStrategy(AbstractPresampleStrategy):
     def __init__(self, config: dict, modyn_config: dict, pipeline_id: int, maximum_keys_in_memory: int):
         super().__init__(config, modyn_config, pipeline_id, maximum_keys_in_memory)
 
-        if "downsampled_batch_size" not in self._config:
-            raise ValueError("To use downsampling strategies, you have to specify the downsampled_batch_size")
-        self.downsampled_batch_size = self._config["downsampled_batch_size"]
+        if "sample_then_batch" not in config:
+            raise ValueError(
+                "Please specify if you want to sample and then batch or vice versa. "
+                "Use the sample_then_batch parameter"
+            )
+        if config["sample_then_batch"]:
+            self.downsampling_mode = DownsamplingMode.SAMPLE_THEN_BATCH
+        else:
+            self.downsampling_mode = DownsamplingMode.BATCH_THEN_SAMPLE
 
-        if not isinstance(self.downsampled_batch_size, int):
-            raise ValueError("The downsampled batch size must be an integer")
+        if self.downsampling_mode == DownsamplingMode.BATCH_THEN_SAMPLE and "downsampling_period" in config:
+            raise ValueError("downsampling_period can be used only in sample-then-batch.")
+
+        self.downsampling_period = config.get("downsampling_period", 1)
+
+        if "downsampled_batch_ratio" not in config:
+            raise ValueError("Please specify downsampled_batch_ratio to use downsampling methods")
```

Comment:
Yes, the schema had a typo (because previously the ratio was only for STB). I changed and renamed to downsampling_ratio (so the two parameters are downsampling_ratio and presampling_ratio, both integers between 0 and 100)